crates/starknet-devnet/tests/test_memory_test.rs

```diff
@@ -0,0 +1,47 @@
+pub mod common;
+
+mod memory_test {
+
+    use peak_alloc::PeakAlloc;
+
+    use crate::common::background_devnet::BackgroundDevnet;
+
+    #[global_allocator]
+    static PEAK_ALLOC: PeakAlloc = PeakAlloc;
+
+    static DUMMY_ADDRESS: u128 = 1;
+    static DUMMY_AMOUNT: u128 = 1;
+
+    async fn mint_iter(f: &str) -> f32 {
+        // reset peak usage
+        PEAK_ALLOC.reset_peak_usage();
+
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&["--state-archive-capacity", f])
+            .await
+            .expect("Could not start Devnet");
+
+        for n in 1..=5_000 {
+            println!("n: {:?}", n);
+            let mint_tx_hash = devnet.mint(DUMMY_ADDRESS, DUMMY_AMOUNT).await;
+            println!("mint_tx_hash: {:?}", mint_tx_hash);
+        }
+
+        // set and return peak usage after recurring mint run
+        PEAK_ALLOC.peak_usage_as_mb()
+    }
+
+    #[tokio::test]
+    async fn test_memory_limit() {
+        // tolerance limit of 0.1%
+        const TOLERANCE: f32 = 0.001;
+        let full = mint_iter("full").await;
+        let none = mint_iter("none").await;
+
+        let diff = (full - none) / none;
+        println!("Diff: {}", diff);
+        println!("Full mem in MB: {}", full);
+        println!("Full mem in MB: {}", none);
+
+        assert!(diff < TOLERANCE, "memory difference should not exceed more than {}% in tolerance. Current Difference {}% between full state archive in {}MB and no state archive in {}MB", TOLERANCE * 100.0, diff * 100.0, full, none);
```

Comment:
I would expect this assertion to fail, but on my computer (an x86_64 machine) it passed, having printed the following:
```
Diff: -0.4698299
Full mem in MB: 1.7984352
Full mem in MB: 3.3921852
```
The memory usage seems to not be read correctly, since by looking at `htop` output I see the total system memory usage exceeding 15 GB, and here it's mere 3 MB.

Reply:
Add any test that will actually measure the memory used, because the current test is not doing that (for sure the program is not taking up 3 MB, when it's actually gigabytes).

Reply:
dhat isn't any better unfortunately. I've explored more tools but I cannot find any that can examine the entire physical memory during an automated test.

Reply:
If you want to increase the chance of your PR getting accepted, you need to address what its corresponding issue initially requested. If you can't detect actual memory growth with a test/script, then I'm not so sure about the future of this PR.

Reply:
But still, your feedback on this matter has been valuable, so even if this PR does not get merged, you can expect some kind of reward.

Reply:
These smaller libraries just detect heap allocations rather than full memory usage. The memory growth would need to be inspected manually using an OS specific tool like valgrind, leaks, etc.